Lib/pstats.py

```diff
@@ -776,4 +803,41 @@ def postcmd(self, stop, line):
     except KeyboardInterrupt:
         pass
 
+
+def _build_minimal_path_by_full_path(paths):
+    if not paths:
+        return paths
+
+    completed = {
+        full_path: None
+        for full_path in paths
+    }
+    split_path_by_full = {full_path: Path(full_path).parts for full_path in paths}
+    max_step = max(len(x) for x in split_path_by_full.values())
+    step = 1
+    while step <= max_step:
+        short_path_by_full = {
+            full_path: split_path_by_full[full_path][-step:]
+            for full_path, value in completed.items()
+            if value is None
+        }
+        full_path_by_short = {v: k for k, v in short_path_by_full.items()}
+
+        for short_path, count in Counter(short_path_by_full.values()).most_common():
+            if count == 1:
+                joined_short_path = join(*short_path)
+                # __init__.py is handled specially because it's a very common
+                # file name which gives no clue what file is meant
+                if joined_short_path == '__init__.py':
```

Comment:
I insist on not having special cases for `__init__.py`

Reply:
Why though?